### PR TITLE
Refactor ship before introducing save summary feature

### PR DESCRIPTION
**Pull request by imsdu**

### CODE REVIEW

File: ship/src/main/scala/ch/epfl/bluebrain/nexus/ship/RunShip.scala [hunk 1]
```diff
@@ -12,46 +10,34 @@ import ch.epfl.bluebrain.nexus.delta.sdk.projects.model.ApiMappings
...
-
-  private val logger = Logger[RunShip]
-
-  def loadConfig(config: Option[Path]): IO[ShipConfig]
-
-  def eventsStream(path: Path, fromOffset: Offset): Stream[IO, RowEvent]
+object RunShip {
 
-  def run(path: Path, config: Option[Path], fromOffset: Offset = Offset.start): IO[ImportReport] = {
+  def apply(eventsStream: Stream[IO, RowEvent], config: InputConfig, xas: Transactors): IO[ImportReport] = {
```
Comment (imsdu) [inline ship/src/main/scala/ch/epfl/bluebrain/nexus/ship/RunShip.scala:24]: `RunShip` does not have to be aware of the config and if we run on S3 or not

File: ship/src/main/scala/ch/epfl/bluebrain/nexus/ship/config/ShipConfig.scala [hunk 2]
```diff
@@ -3,41 +3,24 @@ package ch.epfl.bluebrain.nexus.ship.config
...
-    baseUri: BaseUri,
-    database: DatabaseConfig,
-    S3: S3Config,
-    eventLog: EventLogConfig,
-    organizations: OrganizationCreationConfig,
-    projectMapping: ProjectMapping = Map.empty,
-    viewDefaults: ViewDefaults,
-    serviceAccount: ServiceAccountConfig
-)
+final case class ShipConfig(database: DatabaseConfig, s3: S3Config, input: InputConfig)
```
Comment (imsdu) [inline ship/src/main/scala/ch/epfl/bluebrain/nexus/ship/config/ShipConfig.scala:20]: Separating purely database and s3 config from the others

File: ship/src/test/scala/ch/epfl/bluebrain/nexus/ship/S3RunShipSuite.scala [hunk 3]
```diff
@@ -1,78 +1,51 @@
...
 import munit.AnyFixture
-import org.scalatest.time.SpanSugar.convertIntToGrainOfTime
-import software.amazon.awssdk.services.s3.model.{CreateBucketRequest, PutObjectRequest, PutObjectResponse}
 
-import java.nio.file.Paths
-import scala.concurrent.duration.Duration
+import scala.concurrent.duration.{Duration, DurationInt}
 
-class S3RunShipSuite extends NexusSuite with RunShipSuite.Fixture with LocalStackS3StorageClient.Fixture {
+class S3RunShipSuite
```
Comment (imsdu) [inline ship/src/test/scala/ch/epfl/bluebrain/nexus/ship/S3RunShipSuite.scala:17]: In a later PR, we could change this test to limit it to read the file and not run the import itself

**Review by olivergrabinski — APPROVED** [review]

File: delta/plugins/storage/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/LocalStackS3StorageClient.scala [hunk 4]
```diff
@@ -8,14 +8,27 @@ import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.DigestAlgori
...
 import munit.catseffect.IOFixture
 import org.testcontainers.containers.localstack.LocalStackContainer.Service
+import software.amazon.awssdk.services.s3.model.{CreateBucketRequest, PutObjectRequest, PutObjectResponse}
+
+import java.nio.file.Paths
 
 object LocalStackS3StorageClient {
   val ServiceType = Service.S3
 
+  def uploadFileToS3(s3Client: S3AsyncClientOp[IO], bucket: BucketName, path: Path): IO[PutObjectResponse] = {
```
Comment (dantb) [inline delta/plugins/storage/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/LocalStackS3StorageClient.scala:24]: Could go in `S3Helpers` (might be opportunities for reuse)

File: delta/testkit/src/main/scala/ch/epfl/bluebrain/nexus/testkit/minio/LocalStackS3.scala [hunk 5]
```diff
@@ -10,11 +10,12 @@ import software.amazon.awssdk.services.s3.S3AsyncClient
 
 object LocalStackS3 {
   val ServiceType = Service.S3
+  val Version     = "3.2"
```
[the remark this answers is not in the document]
Comment (imsdu) [reply delta/testkit/src/main/scala/ch/epfl/bluebrain/nexus/testkit/minio/LocalStackS3.scala:13]: We already do this for Postgres / ES / Keycloak
Docker images are dependencies and as other dependencies, it is better to control their version than waking up and having a new version of localstack not compatible with testcontainers for example.

File: delta/sourcing-psql/src/test/scala/ch/epfl/bluebrain/nexus/delta/sourcing/postgres/Doobie.scala [hunk 6]
```diff
@@ -34,9 +36,41 @@ object Doobie {
...
   trait Fixture { self: NexusSuite =>
-    val doobie: catseffect.IOFixture[Transactors] =
+
+    val doobie: IOFixture[Transactors] =
       ResourceSuiteLocalFixture("doobie", resource(PostgresUser, PostgresPassword))
 
+    /**
+      * Truncate all tables after each test
+      */
+    val doobieTruncateAfterTest: IOFixture[Transactors] = new IOFixture[Transactors]("doobie") {
```
Comment (dantb) [inline delta/sourcing-psql/src/test/scala/ch/epfl/bluebrain/nexus/delta/sourcing/postgres/Doobie.scala:46]: shame you have to manually roll this, would be nice to be able to configure the fixture more granularly

**Review by dantb — APPROVED** [review]

**Pull request merged**